packages/teams-js/src/internal/utils.ts

```diff
@@ -39,6 +40,13 @@ function validateHostAgainstPattern(pattern: string, host: string): boolean {
 
 const validateOriginLogger = getLogger('validateOrigin');
 
+function validateOriginHandler(messageOrigin: URL): Promise<boolean> {
+  return new Promise<boolean>((resolve) => {
+    const messageOriginUrl = JSON.parse(JSON.stringify(messageOrigin));
```

Comment:
>JSON.parse(JSON.stringify(messageOrigin)); [](http://example.com/codeflow?start=29&length=42)

What is the goal of this line? It's not obvious to someone reading it why we would stringify an object and then immediately parse it.

Reply:
Thanks for updating this to have a clearer name (`sanitizeAndStrigifyURL`). Can you say more about how doing this roundtrip sanitizes the URL? I couldn't think of a way it was doing that but I'm not an expert in this type of thing. I asked ChatGPT to explain the code as well and it seemed unclear there, too (though AI can obviously be wrong). Do you have an example of something this prevents/fixes?

<img width="554" alt="image" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/36546967/0914d818-57ad-41b1-bae2-f073f457e7e1">

Reply:
Sorry, ChatGPT is actually right in this situation. I'm planning to use this method as a sanitize function for the url eventually. Its current iteration simply takes a URL object and then removes any associated functions attached to it so that it is able to sent as a message, otherwise it would be considered an invalid object and throw an error. This was purely used for backwards compatibility testing of the validation methods on the Host side to ensure parity with the current methods. 

The next iteration would change the `sendAndUnwrap('validateDomains`)` call to just use the host string.